apps/aecore/lib/aecore/peers/worker.ex

```diff
@@ -109,17 +123,28 @@ defmodule Aecore.Peers.Worker do
     {:reply, peers, peers}
   end
 
-  def handle_cast({:broadcast_tx, tx}, peers) do
-    serialized_tx = 
-    Serialization.tx(tx, :serialize)
-    |> Poison.encode!()
-    for peer <- peers do
-      Client.send_tx(peer, serialized_tx)
-    end
+  ## Async operations
 
+  def handle_cast({:async_request, {type, data}}, peers) do
+    send_to_peers(:post, type, data, Map.keys(peers))
     {:noreply, peers}
   end
+
   def handle_cast(_any, peers) do
-    {:noreply, peers}
+    Logger.info("[Peers] Unhandled cast message:  #{inspect(peers)}")
+  end
+
+  ## Internal functions
+  defp send_to_peers(:post, uri, data, []) do
+    Logger.warn("Empty peers list")
+  end
+  defp send_to_peers(:post, uri, data, peers) do
```

Comment:
sending will always happen with `:post`, no need to specify. What is `uri` needed for, each peer should have its own uri

Reply:
https://github.com/aeternity/elixir-research/blob/GH-75/apps/aehttpclient/lib/client.ex#L37

Reply:
I have to look at it in more detail, thanks

Reply:
Can you rename it to something else? In the peers worker, `uri` is always used as the address of the peer, this is why I was so confused